Doctrine/Phpcr/AutoRouteListener.php

```diff
@@ -73,6 +74,13 @@ public function onFlush(ManagerEventArgs $args)
                     $dm->persist($autoRoute);
                     $uow->computeChangeSets();
                 }
+
+                // reset locale to the original locale
+                if (null !== $locale) {
```

Comment:
As lukas mentioned `findTranslation` was caching the items, I was a bit unsure what would be better, performance-wise.